Code/Tools/SceneAPI/SceneBuilder/Importers/AssImpBlendShapeImporter.cpp

```diff
@@ -80,41 +82,102 @@ namespace AZ
+                            {
+                                AZ_Error(
+                                    Utilities::ErrorWindow, false,
+                                    "Meshes %s and %s on node %s have mismatched animations %s and %s at index %d. This can be resolved by "
+                                    "either manually separating meshes by material in the source scene file, or by updating this logic to "
+                                    "handle out of order animation indices.",
+                                    firstExistingAnim.m_aiMesh->mName.C_Str(),
+                                    aiMesh->mName.C_Str(),
+                                    context.m_sourceNode.GetName(),
+                                    firstExistingAnim.m_aiAnimMesh->mName.C_Str(),
+                                    aiAnimMesh->mName.C_Str(), animIdx);
+                                return Events::ProcessingResult::Failure;
+                            }
+                        }
+
+                        animMeshIndexToSceneMeshes[animIdx].m_animMeshAndSceneMeshIndex.emplace_back(
+                            AnimMeshAndSceneMeshIndex(aiAnimMesh, aiMesh));
                     }
                 }
 
-                for (const auto& animToMeshIndex : animToMeshToAnimMeshIndices)
+                for (const auto& animMeshToSceneMeshes : animMeshIndexToSceneMeshes)
                 {
                     AZStd::shared_ptr<SceneData::GraphData::BlendShapeData> blendShapeData =
                         AZStd::make_shared<SceneData::GraphData::BlendShapeData>();
 
+                    if (animMeshToSceneMeshes.second.m_animMeshAndSceneMeshIndex.size() == 0)
+                    {
+                        AZ_Error(Utilities::ErrorWindow, false, "Blend shape nimations were expected but missing on node %s.",
+                            context.m_sourceNode.GetName());
+                        return Events::ProcessingResult::Failure;
+                    }
                     // Some DCC tools, like Maya, include a full path separated by '.' in the node names.
                     // For example, "cone_skin_blendShapeNode.cone_squash"
                     // Downstream processing doesn't want anything but the last part of that node name,
                     // so find the last '.' and remove anything before it.
-                    AZStd::string nodeName(animToMeshIndex.first);
+                    AZStd::string nodeName(animMeshToSceneMeshes.second.m_animMeshAndSceneMeshIndex[0].m_aiAnimMesh->mName.C_Str());
                     size_t dotIndex = nodeName.rfind('.');
                     if (dotIndex != AZStd::string::npos)
                     {
                         nodeName.erase(0, dotIndex + 1);
                     }
                     int vertexOffset = 0;
                     RenamedNodesMap::SanitizeNodeName(nodeName, context.m_scene.GetGraph(), context.m_currentGraphPosition, "BlendShape");
```

Comment:
just curious why "SanitizeNodeName" does not do the rfind(.) + erase

Reply:
The comments in SanitizeNodeName explains how it handles it there. It's because the original use cases SanitizeNodeName was created for use the dot for hierarchy separation, parent.child, and replaces dots with underscores.

In the case of animations, we wanted to strip that beginning part entirely, we don't want the parent's name, we just want the animation name.